sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -291,7 +291,8 @@ private[sql] class HDFSBackedStateStoreProvider extends StateStoreProvider with
       newMap
     }
     catch {
-      case e: SparkException if e.getCondition.contains("CANNOT_LOAD_STATE_STORE") =>
+      case e: SparkException
+        if Option(e.getCondition).contains("CANNOT_LOAD_STATE_STORE") =>
```

Comment:
Good catch

https://github.com/apache/spark/blob/f0f850dc21ecf928995736daf8b790082e099221/sql/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala#L3151-L3160

Do you think there is also a risk of `NPE` in the code above? In the current production code, `e.getCondition` is called over 35 times. Are all other invocations free from the risk of NPE?

Reply:
Thanks for the flagging!

Because of how a StreamingQueryException is constructed:
https://github.com/apache/spark/blob/331d0bf30092be62191476e4a679b403e1a369b9/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala#L368

`getCondition` (former `getErrorClass`) from a `StreamingQueryException` should always return non-null. So I believe there should never be an NPE thrown.